src/CoreBundle/DataModel/CmsMasterPagdef.php

```diff
@@ -17,11 +17,23 @@ class CmsMasterPagdef
      */
     private $layoutFile;
 
-    public function __construct(string $id, array $moduleList, string $layoutFile)
+    /**
+     * @var \TdbCmsRight[]
+     */
+    private $allowedRights;
+
+    /**
+     * @param string         $id
+     * @param array          $moduleList
+     * @param string         $layoutFile
+     * @param \TdbCmsRight[] $allowedRights
+     */
+    public function __construct(string $id, array $moduleList, string $layoutFile, array $allowedRights)
```

Comment:
I would suggest changing the name of `allowedRights` into `cmsRightAllowList` everywhere

Reply:
oh - and this is the data model for alle pagedefs, right? Including the pagedefs for the frontend... If that is the case, then again we should avoid introducing a parameter that will be used only by a very small fraction of pages.

I would suggest extending the class to create a backend specific version.

`class BackedPageDefDataModel extends CmsMasterPagdef`

The extension can then add the attributes required for backend page handling.

Reply:
From an information encapsulation point this probably makes sense.

However as interfaces use this as return type (ie DataAccessCmsMasterPagedefInterface::get()) we would need at least at some points an instanceof (or a hard cast) where the else case would be a runtime error.

Reply:
Add a special "file version" of the data model - and check for it in two locations.